bigquery/main_test.go

```diff
@@ -60,15 +60,20 @@ func (ml *fakeLayer) MediaType() (types.MediaType, error) {
 	return "", nil
 }
 
-type mockBQ struct {
+type fakeBQ struct {
 	validSchema bool
+	WrittenRows []*bqRow
 }
 
-type mockBQFactory struct {
+type fakeBQFactory struct {
+	MaybeFake *fakeBQ
 }
 
-func (bqf *mockBQFactory) Make(ctx context.Context) (bq, error) {
-	return &mockBQ{}, nil
+func (bqf *fakeBQFactory) Make(ctx context.Context) (bq, error) {
+	if bqf.MaybeFake != nil {
```

Comment:
Up to you: just simplify this to just return the `maybeFake`, make tests always set this field. Then you can remove the nil check and the `maybe` prefix.